core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -1807,6 +1817,7 @@ class SocketServerTest {
           currentPollValues ++= newValues
         } else
           deferredValues ++= newValues
+        newValues.clear()
```

Comment:
What is the goal of this?

Reply:
The reason I was asking is that we call `toBuffer` before calling this method for a couple of cases. That will basically create a copy, so then this `clear` won't do anything useful?

Reply:
ah, I refactored a bit to clear the original buffer in each case.